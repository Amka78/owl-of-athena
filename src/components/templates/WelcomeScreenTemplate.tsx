//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import {
    Button,
    ContentText,
    ContentTextProps,
    ContentTitle,
    ContentTitleProps,
    FlexSpacer,
    StandardView,
} from "../atoms";
import { Message, MessageKeys } from "../../constants";
import { useLocale } from "../../hooks";
import { TemplateButtonProps } from "./TempatedProps";
//#endregion

//#region Types
export type WelcomeScreenTemplateProps = {
    contentTitle?: ContentTitleProps;
    contentText?: ContentTextProps;
    standaloneButton: TemplateButtonProps;
    loginButton: TemplateButtonProps;
    signupButton: TemplateButtonProps;
    locale?: string;
};
//#endregion

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    useLocale(props.locale);
    return (
        <StandardView>
            <FlexSpacer></FlexSpacer>
            <ContentTitle {...props.contentTitle} style={{ flex: 1 }}>
                {Message.get(MessageKeys.welcome_title)}
            </ContentTitle>
            <ContentText {...props.contentText} style={{ flex: 1 }}>
                {Message.get(MessageKeys.welcome_text)}
            </ContentText>
            <FlexSpacer></FlexSpacer>
            <View>
                <Button {...props.standaloneButton}>
                    {Message.get(MessageKeys.welcome_standalone_button)}
                </Button>
                <Button {...props.loginButton}>
                    {Message.get(MessageKeys.welcome_login_button)}
                </Button>
                <Button {...props.signupButton}>
                    {Message.get(MessageKeys.welcome_signup_button)}
                </Button>
            </View>
        </StandardView>
    );
};
//#endregion
