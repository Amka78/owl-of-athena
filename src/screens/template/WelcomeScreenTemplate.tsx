//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import {
    Button,
    ContentText,
    ContentTitle,
    StandardView,
} from "../../components";
import { MessageLocalizationParam } from "../../constants/Message";
//#endregion

export type WelcomeScreenTemplateProps = {
    contentTitleText: MessageLocalizationParam;
    contentText: MessageLocalizationParam;
    standaloneButtonText: MessageLocalizationParam;
    standaloneButtonPress: () => void;
    loginButtonText: MessageLocalizationParam;
    loginButtonPress: () => void;
    signupButtonText: MessageLocalizationParam;
    signupButtonPress: () => void;
};

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    return (
        <StandardView>
            <ContentTitle>{props.contentTitleText}</ContentTitle>
            <ContentText>{props.contentText}</ContentText>
            <View>
                <Button onPress={props.standaloneButtonPress}>
                    {props.standaloneButtonText}
                </Button>
                <Button onPress={props.loginButtonPress}>
                    {props.loginButtonText}
                </Button>
                <Button onPress={props.signupButtonPress}>
                    {props.signupButtonText}
                </Button>
            </View>
        </StandardView>
    );
};
//#endregion
