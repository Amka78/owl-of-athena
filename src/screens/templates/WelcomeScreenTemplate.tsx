//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import {
    Button,
    ButtonProps,
    ContentText,
    ContentTextProps,
    ContentTitle,
    ContentTitleProps,
    StandardView,
} from "../../components";
//#endregion

//#region Types
export type WelcomeScreenTemplateProps = {
    contentTitle: ContentTitleProps;
    contentText: ContentTextProps;
    standaloneButton: ButtonProps;
    loginButton: ButtonProps;
    signupButton: ButtonProps;
};
//#endregion

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    return (
        <StandardView>
            <ContentTitle {...props.contentTitle}></ContentTitle>
            <ContentText {...props.contentText}></ContentText>
            <View>
                <Button {...props.standaloneButton}></Button>
                <Button {...props.loginButton}></Button>
                <Button {...props.signupButton}></Button>
            </View>
        </StandardView>
    );
};
//#endregion
