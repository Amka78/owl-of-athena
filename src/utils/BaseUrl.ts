import Constants from "expo-constants";

/**
 * Manage API-URL for each environment.
 *
 * @export
 * @class BaseUrl
 */
export default class BaseUrl {
    public static get(relCh?: string) {
        if (!relCh && Constants.manifest.releaseChannel) {
            relCh = Constants.manifest.releaseChannel;
        }

        if (relCh !== undefined) {
            if (relCh!.includes("staging")) {
                return IWinksRestAPI.staging.url;
            } else if (relCh!.includes("prod")) {
                return IWinksRestAPI.prod.url;
            }
        }

        return IWinksRestAPI.dev.url;
    }

    public static getDownloadUrl(
        image: string | string[],
        index?: number
    ): string | undefined {
        if (typeof image === "undefined") {
            return undefined;
        } else if (typeof image === "string") {
            return BaseUrl.get() + image;
        } else {
            if (index === undefined) {
                index = 0;
            }
            return BaseUrl.get() + image![index];
        }
    }
}

export const IWinksRestAPI = {
    dev: {
        url: "https://api.iwinks.io/"
    },
    staging: {
        url: "https://api.iwinks.io/"
    },
    prod: {
        url: "https://api.iwinks.io/"
    }
};
