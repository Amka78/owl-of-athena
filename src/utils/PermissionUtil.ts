import * as Permissions from "expo-permissions";
import { Linking } from "react-native";

export default class PermissionUtil {
    public static async getPermission(permission: any) {
        const { status } = await Permissions.askAsync(permission);
        if (status !== "granted") {
            Linking.openURL("app-settings:");
            return false;
        }
        return true;
    }
}
