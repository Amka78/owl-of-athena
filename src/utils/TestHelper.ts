import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
