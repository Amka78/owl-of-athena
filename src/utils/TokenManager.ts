import { Platform } from "@unimodules/core";
import * as SecureStore from "expo-secure-store";
import { AsyncStorage } from "react-native";

enum SecureStoreKeys {
    JWT_TOKEN = "jwt_token"
}
/**
 * Token Management Class
 *
 * @export
 * @class TokenManager
 */
export default class TokenManager {
    public static async hasToken(): Promise<boolean> {
        const result = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        return result !== undefined;
    }

    /**
     * get token
     *
     * @static
     * @returns {(Promise<string | undefined>)}
     * @memberof TokenManager
     */
    public static async get(): Promise<string | undefined> {
        const token = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        if (token) {
            return token;
        }

        return undefined;
    }

    /**
     * set token
     *
     * @static
     * @param {TokenDto} value
     * @memberof TokenManager
     */
    public static async set(value: string) {
        await this.setPerPlatform(SecureStoreKeys.JWT_TOKEN, value);
    }

    public static async reset() {
        await this.deletePerPlatform(SecureStoreKeys.JWT_TOKEN);
    }

    private static async getPerPlatform(key: string): Promise<string | null> {
        if (Platform.OS === "web") {
            return await AsyncStorage.getItem(key);
        }

        return await SecureStore.getItemAsync(key);
    }

    private static async setPerPlatform(
        key: string,
        value: string
    ): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.setItem(key, value);
        }

        return await SecureStore.setItemAsync(key, value);
    }

    private static async deletePerPlatform(key: string): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.removeItem(key);
        }

        return await SecureStore.deleteItemAsync(key);
    }
}
