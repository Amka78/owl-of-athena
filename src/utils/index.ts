import BaseUrl from "./BaseUrl";
export { BaseUrl };
