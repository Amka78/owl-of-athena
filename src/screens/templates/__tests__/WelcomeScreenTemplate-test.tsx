import "jest-enzyme";
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";
import { WelcomeScreenTemplate } from "../WelcomeScreenTemplate";
import { TestHelper } from "../../../utils/TestHelper";

let component: ShallowWrapper<unknown, unknown, unknown>;

describe("WelcomeScreenTemplate UnitTest", () => {
    it("renders correctly", () => {
        const testKey = "test";
        const testEvent: () => void = () => {
            return;
        };
        component = TestHelper.createMock(
            <WelcomeScreenTemplate
                contentTitle={{ children: testKey }}
                contentText={{ children: testKey }}
                standaloneButton={{ children: testKey, onPress: testEvent }}
                loginButton={{ children: testKey, onPress: testEvent }}
                signupButton={{ children: testKey, onPress: testEvent }}
            ></WelcomeScreenTemplate>
        );

        expect(TestHelper.toJson(component)).toMatchSnapshot();
    });
});
