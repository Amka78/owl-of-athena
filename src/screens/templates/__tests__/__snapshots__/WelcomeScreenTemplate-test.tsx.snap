// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreenTemplate UnitTest renders correctly 1`] = `
<Component>
  <Component>
    test
  </Component>
  <Component>
    test
  </Component>
  <View>
    <Component
      onPress={[Function]}
    >
      test
    </Component>
    <Component
      onPress={[Function]}
    >
      test
    </Component>
    <Component
      onPress={[Function]}
    >
      test
    </Component>
  </View>
</Component>
`;
